app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt

```diff
@@ -522,13 +558,110 @@ class FeedFragment : BaseStateFragment<FeedState>() {
         )
     }
 
+    /**
+     * Highlights all items that are after the specified time
+     */
+    private fun highlightNewItemsAfter(updateTime: OffsetDateTime) {
+        var highlightCount = 0
+
+        var doCheck = true
+
+        for (i in 0 until groupAdapter.itemCount) {
+            val item = groupAdapter.getItem(i) as StreamItem
+
+            var typeface = Typeface.DEFAULT
+            var resid = R.attr.selectableItemBackground
+            if (doCheck) {
+                if (item.streamWithState.stream.uploadDate?.isAfter(updateTime) != false) {
+                    typeface = Typeface.DEFAULT_BOLD
+                    resid = R.attr.dashed_border
+                    highlightCount++
+                } else {
+                    // Increases execution time due to the order of the items (newest always on top)
+                    // Once a item is is before the updateTime we can skip all following items
+                    doCheck = false
+                }
+            }
+
+            // The highlighter has to be always set
+            // When it's only set on items that are highlighted it will highlight all items
+            // due to the fact that itemRoot is getting recycled
+            item.execBindEnd = Consumer { viewBinding ->
+                val context = viewBinding.itemRoot.context
+                viewBinding.itemRoot.background =
+                    androidx.core.content.ContextCompat.getDrawable(
+                        context,
+                        android.util.TypedValue().apply {
+                            context.theme.resolveAttribute(
+                                resid,
+                                this,
+                                true
+                            )
+                        }.resourceId
+                    )
```

Comment:
This removes the selectable item background though. Are we sure this is what we want?

Reply:
I think you can merge two drawables together, but idk if it works in older android versions. If it turns out that can't be done, just add a `View` that spans the whole item with the desired background

Reply:
https://stackoverflow.com/a/45153629 4tw.

Should be fixed now:

![NP7050FixSelectHighlight](https://user-images.githubusercontent.com/40789489/135156209-7299af5b-7cc0-46e4-8fce-5b764a1274a3.gif)

@Stypox Please recheck the affected code.